vernac/attributes.ml

```diff
@@ -159,12 +160,16 @@ let bool_attribute ~name : bool option attribute =
   key_value_attribute ~key:name ~default:true ~values
 
 (* Variant of the [bool] attribute with only two values (bool has three). *)
+let qualid_is_this_ident fp id =
+  Libnames.qualid_is_ident fp &&
+  Names.Id.to_string @@ Libnames.qualid_basename fp = id
+
 let get_bool_value ?loc ~key ~default =
   function
   | VernacFlagEmpty -> default
-  | VernacFlagLeaf (FlagIdent "yes") ->
+  | VernacFlagLeaf (FlagQualid q) when qualid_is_this_ident q "yes" ->
     true
-  | VernacFlagLeaf (FlagIdent "no") ->
+  | VernacFlagLeaf (FlagQualid q) when qualid_is_this_ident q "no" ->
```

Comment:
I am not really fond of this approach. What about having a `FlagBool` constructor and considering that `yes` and `no` are necessarily boolean values. (And if the user wants to refer to a `yes` or `no` identifier in an attribute, they can always qualify it.)

Reply:
That's the yaml approach where norway gets turned into false

Reply:
Noway

Reply:
Jokes on the side, I think it should go in another pr

Reply:
It is this PR that is removing `FlagIdent` and adding `FlagQualid`. And it is this PR that now passes `yes` and `no` as qualified identifiers. So, I disagree that it is material for a different PR.

Reply:
My POV is that it is not this pr that decided to use strings as boolean, but it just perpetuates the current very debatable approach.

I'll see if I can push a cleanup commit while waiting for my plane.

Reply:
Done. Not sure the result is as nice as one would expect, since some "parsing facilities" still need to go through strings. I'm not cleaning that up.

Reply:
please revert